staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -228,8 +229,10 @@ func (s *EtcdOptions) Complete(
 	}
 
 	if len(s.EncryptionProviderConfigFilepath) != 0 {
-		ctxTransformers, closeTransformers := wait.ContextForChannel(stopCh)
-		ctxServer, _ := wait.ContextForChannel(stopCh) // explicitly ignore cancel here because we do not own the server's lifecycle
+		baseCtx := wait.ContextForChannel(stopCh)
+		ctxTransformers, closeTransformers := context.WithCancel(baseCtx)
+
+		ctxServer := wait.ContextForChannel(stopCh) // explicitly ignore cancel here because we do not own the server's lifecycle
```

Comment:
Do we actually need this?

We effectively get a copy of baseCtx, but the behave exactly the same way, no?

Reply:
interesting, indeed the `closeTransformer`cancellation should not close teh parent context, right?

Reply:
yes - exactly

Reply:
I’ll fix that up, thanks (had the same thought, but in refactor mode I try to keep myself from changing too much).